### PR TITLE
Feature/api docs

### PR DESCRIPTION
**Pull request by collijk**
- joint value post processor renamed to union post processor.  Reasoning in the doc string.
- Added a 'missing_value_source' type and put pipelines with missing sources explicitly in the dependency graph.  This will be used later for visualization.
- Renamed the interface manager to just '_manager', which i've been doing piecemeal for a while.  I should just do one pr with all of them.  Todo soon.
- Axe the set combiner since it was never used

Everything else is docs

### CODE REVIEW

File: src/vivarium/framework/lookup.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 import pandas as pd
```
Comment (collijk) [inline src/vivarium/framework/lookup.py]: Changes in this file have already been reviewed and approved.

File: src/vivarium/framework/population.py [hunk 2]
```diff
@@ -93,10 +93,11 @@ def get(self, index: pd.Index, query: str = '') -> pd.DataFrame:
         """
         pop = self.manager.get_population(True).loc[index]
 
-        if self._query:
```
Comment (collijk) [inline src/vivarium/framework/population.py]: Changes in this file have already been reviewed and approved.

File: src/vivarium/framework/resource.py [hunk 3]
```diff
@@ -30,7 +30,7 @@ class ResourceError(VivariumError):
     pass
 
 
-RESOURCE_TYPES = {'value', 'value_source', 'value_modifier', 'column', 'stream'}
+RESOURCE_TYPES = {'value', 'value_source', 'missing_value_source', 'value_modifier', 'column', 'stream'}
```
Comment (collijk) [inline src/vivarium/framework/resource.py]: To use later in graph display. We allow missing value sources for pipelines that never get called.

File: tests/framework/test_engine.py [hunk 4]
```diff
@@ -48,7 +48,7 @@ def test_SimulationContext_init_default(components):
     assert isinstance(sim._builder.lookup, LookupTableInterface)
     assert sim._builder.lookup._lookup_table_manager is sim._tables
     assert isinstance(sim._builder.value, ValuesInterface)
-    assert sim._builder.value._value_manager is sim._values
+    assert sim._builder.value._manager is sim._values
```
Comment (collijk) [inline tests/framework/test_engine.py]: Slowly making all the manager names just "manager"

File: src/vivarium/framework/resource.py (hunk 3 above)
Comment (halfhorst) [reply src/vivarium/framework/resource.py]: as in the source of the pipeline is "missing value" ? Is that what that means?

File: src/vivarium/framework/values.py [hunk 5]
```diff
@@ -171,46 +301,86 @@ def register_value_producer(self, value_name, source,
...
+        ----------
+        value_name :
+            The name of the dynamic value pipeline to be modified.
+        modifier :
+            A function that modifies the source of the dynamic value pipeline
+            when called. If the pipeline has a ``replace_combiner``, the
+            modifier should accept the same arguments as the pipeline source
+            with an additional last positional argument for the results of the
+            previous stage in the pipeline. For the ``list_combiner`` and
+            ``set_combiner`` strategies, the pipeline modifiers should have
```
Comment (halfhorst) [inline src/vivarium/framework/values.py]: set was removed

File: src/vivarium/framework/resource.py (hunk 3 above)
Comment (collijk) [reply src/vivarium/framework/resource.py]: This will be a pipeline with no source. Which we allow, I just want to visually represent.

**Pull request merged**